From d59a35bbcea0558c6ce6d764c564e374cde349be Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Sun, 4 Mar 2018 17:20:25 -0700
Subject: [PATCH 125/773] sound: bcm: Fix memset dereference warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This warning appears with GCC 6.4.0 from toolchains.bootlin.com:

../sound/soc/bcm/allo-piano-dac-plus.c: In function ‘snd_allo_piano_dac_init’:
../sound/soc/bcm/allo-piano-dac-plus.c:711:30: warning: argument to ‘sizeof’ in ‘memset’ call is the same expression as the destination; did you mean to dereference it? [-Wsizeof-pointer-memaccess]
  memset(glb_ptr, 0x00, sizeof(glb_ptr));
                              ^

Suggested-by: Phil Elwell <phil@raspberrypi.org>
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 sound/soc/bcm/allo-piano-dac-plus.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/sound/soc/bcm/allo-piano-dac-plus.c
+++ b/sound/soc/bcm/allo-piano-dac-plus.c
@@ -706,11 +706,10 @@ static int snd_allo_piano_dac_init(struc
 	struct snd_soc_card *card = rtd->card;
 	struct glb_pool *glb_ptr;
 
-	glb_ptr = kmalloc(sizeof(struct glb_pool), GFP_KERNEL);
+	glb_ptr = kzalloc(sizeof(struct glb_pool), GFP_KERNEL);
 	if (!glb_ptr)
 		return -ENOMEM;
 
-	memset(glb_ptr, 0x00, sizeof(glb_ptr));
 	card->drvdata = glb_ptr;
 	glb_ptr->dual_mode = 2;
 	glb_ptr->set_mode = 0;
